common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/LwM2mTransportContextServer.java

```diff
@@ -159,4 +159,118 @@ public ObjectModel parseFromXmlToObjectModel(byte[] xmlByte, String streamName,
+                                .setStringV((String) resultsResourceValue.getValue()).build();
+                    case LONG:
+                        return TransportProtos.KeyValueProto.newBuilder()
+                                .setKey(resultsResourceValue.getResourceName())
+                                .setType(TransportProtos.KeyValueType.LONG_V)
+                                .setLongV((Long) resultsResourceValue.getValue()).build();
+                    case DOUBLE:
+                        return TransportProtos.KeyValueProto.newBuilder()
+                                .setKey(resultsResourceValue.getResourceName())
+                                .setType(TransportProtos.KeyValueType.DOUBLE_V)
+                                .setDoubleV((Double) resultsResourceValue.getValue()).build();
+                    case JSON:
+                        return TransportProtos.KeyValueProto.newBuilder()
+                                .setKey(resultsResourceValue.getResourceName())
+                                .setType(TransportProtos.KeyValueType.JSON_V)
+                                .setJsonV((String) resultsResourceValue.getValue()).build();
+                    default:
+                        return null;
+                }
+        }
+
+    public DataType getDataTypeEqualsToResourceModelType(ResourceModel.Type resourceType, String resourcePath, boolean isMultiInstances) {
+        if (isMultiInstances) {
+           return DataType.JSON;
+        }
+        else {
+            switch (resourceType) {
+                case BOOLEAN:
+                    return DataType.BOOLEAN;
+                case STRING:
+                case TIME:
+                case OPAQUE:
+                case OBJLNK:
+                    return DataType.STRING;
+                case INTEGER:
+                    return DataType.LONG;
+                case FLOAT:
+                    return DataType.DOUBLE;
+                default:
+            }
+        }
+        throw new CodecException("Invalid DataType for resource %s, got %s, ", resourcePath, resourceType);
+    }
+
+    /**
+     *
+     * @param currentType
+     * @param resourcePath
+     * @return
+     */
+    public ResourceModel.Type getResourceModelTypeEqualsKvProtoValueType(ResourceModel.Type currentType, String resourcePath) {
+        switch (currentType) {
+            case BOOLEAN:
+                return ResourceModel.Type.BOOLEAN;
+            case STRING:
+            case TIME:
+            case OPAQUE:
+            case OBJLNK:
+                return ResourceModel.Type.STRING;
+            case INTEGER:
+                return ResourceModel.Type.INTEGER;
+            case FLOAT:
+                return ResourceModel.Type.FLOAT;
+            default:
+        }
+        throw new CodecException("Invalid ResourceModel_Type for resource %s, got %s", resourcePath, currentType);
+    }
+
+    public Object getValueFromKvProto (TransportProtos.KeyValueProto kv) {
```

Comment:
Method not used?

Reply:
src/main/java/org/thingsboard/server/transport/lwm2m/server/LwM2mTransportServiceImpl.java